### PR TITLE
Add Shell experience docs

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

- Fixes [AB#4298](https://dev.azure.com/content-learn/cebd7ef5-4282-448b-9701-88c8637581b7/_workitems/edit/4298)
- Fixes [AB#4299](https://dev.azure.com/content-learn/cebd7ef5-4282-448b-9701-88c8637581b7/_workitems/edit/4299)
- Fixes [AB#4300](https://dev.azure.com/content-learn/cebd7ef5-4282-448b-9701-88c8637581b7/_workitems/edit/4300)
- Fixes [AB#4306](https://dev.azure.com/content-learn/cebd7ef5-4282-448b-9701-88c8637581b7/_workitems/edit/4306)
- Fixes [AB#4354](https://dev.azure.com/content-learn/cebd7ef5-4282-448b-9701-88c8637581b7/_workitems/edit/4354)

Add Shell experience docs
- Add Overview - optimize-shell.md
- Add tab-completion.md
- Add using-predictors.md
- Add dynamic-help.md
- Rename Compatibility-Aliases.md to using-aliases.md and add more content
- Update toc.yml

## PR Checklist

- [x] **Descriptive Title:** This PR's title is a synopsis of the changes it proposes.
- [x] **Summary:** This PR's summary describes the scope and intent of the change.
- [x] **Contributor's Guide:** I have read the [contributors guide][contrib].
- [x] **Style:** This PR adheres to the [style guide][style].

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[style]: https://docs.microsoft.com/powershell/scripting/community/contributing/powershell-style-guide

### CODE REVIEW

File: reference/docs-conceptual/learn/shell/using-aliases.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+---
+# Using aliases
+
+An alias is an alternate name or shorthand name for a cmdlet or for a command element, such as a
+function, script, file, or executable file. You can run the command using the alias instead of the
+executable name.
+
+## Managing command aliases
+
+PowerShell provide cmdlets for managing command aliases.
```
Comment (michaeltlombardi) [inline reference/docs-conceptual/learn/shell/using-aliases.md:14]:
```suggestion
PowerShell provides cmdlets for managing command aliases.
```

File: reference/docs-conceptual/learn/shell/using-aliases.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+Cmdlet      Import-Alias 7.0.0.0 Microsoft.PowerShell.Utility
+Cmdlet      New-Alias    7.0.0.0 Microsoft.PowerShell.Utility
+Cmdlet      Remove-Alias 7.0.0.0 Microsoft.PowerShell.Utility
+Cmdlet      Set-Alias    7.0.0.0 Microsoft.PowerShell.Utility
+```
+
+For more information, see [about_Aliases][about_Aliases].
+
+Use the [Get-Alias][Get-Alias] cmdlet to list the aliases available in your environment. To list the
+aliases for a single cmdlet, use the **Definition** parameter and specify the executable name.
```
Comment (michaeltlombardi) [inline reference/docs-conceptual/learn/shell/using-aliases.md:34]: Executable is ambiguous here, I think.
```suggestion
aliases for a single cmdlet, use the **Definition** parameter and specify the command name.
```

File: reference/docs-conceptual/learn/shell/using-aliases.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+For more information, see the [Alias attribute][alias-attr] documentation.
+
+In addition to parameter aliases, PowerShell lets you specify the parameter name using the fewest
+characters needed to uniquely identify the parameter. For example, the `Get-ChildItem` cmdlet has
+the **Recurse** and **ReadOnly** parameters. To uniquely identify the **Recurse** parameter you only
+need to provide `-rec`. If you combine that with the command alias, `Get-ChildItem -Recurse` can be
+shortened to `dir -rec`.
+
+## Don't use aliases in script
```
Comment (michaeltlombardi) [inline reference/docs-conceptual/learn/shell/using-aliases.md:111]:
```suggestion
## Don't use aliases in scripts
```

File: reference/docs-conceptual/learn/shell/using-predictors.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+## Using Predictive IntelliSense
+
+When Predictive IntelliSense is enabled, the prediction suggestion appears as colored text following
+the user's cursor. The suggestions from Predictive IntelliSense help new and experienced users of
+PowerShell discover, edit, and execute full commands based on matching predictions. Suggestions can
+come from the user's history and additional domain specific plugins.
+
+![Inline view of a prediction](media/using-predictors/predictor-inline-1.png)
+
+The previous images shows the default `InlineView` of the suggestion. Pressing <kbd>RightArrow</kdb>
```
Comment (michaeltlombardi) [inline reference/docs-conceptual/learn/shell/using-predictors.md:24]:
```suggestion
The previous images shows the default `InlineView` of the suggestion. Pressing <kbd>RightArrow</kbd>
```

image

File: reference/docs-conceptual/learn/shell/using-predictors.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+### Changing keybindings
+
+PSReadLine contains functions to navigate and accept predictions. For example:
+
+- `AcceptSuggestion` - Accept the current inline suggestion
+- `AcceptNextSuggestionWord` - Accept the next word of the inline suggestion
+- `AcceptSuggestion` is built within `ForwardChar`, which is bound to
+  <kdb>RightArrow</kbd> by default
+- `AcceptNextSuggestionWord` is built within the function `ForwardWord`, which can be bound to
+  <kbd>Ctrl</kbd>+<kbd>f</kbd>
```
Comment (michaeltlombardi) [inline reference/docs-conceptual/learn/shell/using-predictors.md:112]: Can be bound or is bound by default?

File: reference/docs-conceptual/learn/shell/dynamic-help.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+![Full screen Dynamic help](./media/dynamic-help/dynamic-help.gif)
+
+The pager in **PSReadLine** allows you to scroll the displayed help using the up and down arrow
+keys. Pressing <kbd>Q</kbd> exits the alternative screen buffer and returns to the current cursor
+position on the command line on the primary screen.
+
+## Focused parameter help
+
+Pressing <kbd>Alt</kbd>-<kbd>h</kbd> provides dynamic help for parameters. The help is shown below
+the current command line similar to `MenuComplete`. The cursor must be at the end of the fully
```
Comment (StevenBucher98) [inline reference/docs-conceptual/learn/shell/dynamic-help.md:31]: May be useful to link docs to `MenuComplete` as I often forget it is a feature and did not connect the similarities without googling it

Comment (sdwheeler) [reply reference/docs-conceptual/learn/shell/dynamic-help.md:31]: I will link this to the menu complete example (with gif) in the other article.

File: reference/docs-conceptual/learn/shell/using-predictors.md (hunk 5 above)
Comment (sdwheeler) [reply reference/docs-conceptual/learn/shell/using-predictors.md:112]: Can be. Not bound by default.

File: reference/docs-conceptual/learn/shell/dynamic-help.md (hunk 6 above)
Comment (sdwheeler) [reply reference/docs-conceptual/learn/shell/dynamic-help.md]: Done

File: reference/docs-conceptual/learn/shell/using-aliases.md (hunk 2 above)
Comment (michaeltlombardi) [reply reference/docs-conceptual/learn/shell/using-aliases.md:34]: That makes sense, resolving.

File: reference/docs-conceptual/learn/shell/using-predictors.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+title: Using predictors is PSReadLine
+ms.date: 7/14/2022
+---
+# Using predictors is PSReadLine
+
+**PSReadLine** 2.1.0 introduced the **Predictive IntelliSense** feature. Predictive IntelliSense
+provides suggestions for full commands based on items from your **PSReadLine** history.
+**PSReadLine** 2.2.2 extends the power of Predictive IntelliSense by adding support for plug-in
+modules that use advanced logic to provide suggestions for full commands. The latest version,
+**PSReadLine** 2.2.6, enables predictions by default.
```
Comment (StevenBucher98) [inline reference/docs-conceptual/learn/shell/using-predictors.md:13]: Enables history based suggestions by default, not the plugin ones.

File: reference/docs-conceptual/learn/shell/using-predictors.md [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+[Announcing General Availability of Az.Tools.Predictor][azpredictor].
+
+The **CompletionPredictor** module adds an IntelliSense experience for anything that can be
+tab-completed in PowerShell. With **PSReadLine** set to `InlineView`, you get the normal tab
+completion experience. When you switch to `ListView`, you get the IntelliSense experience.
+
+![PowerShell IntelliSense using the CompletionPredictor](media/using-predictors/completion-predictor.gif)
+
+You can install the [CompletionPredictor][CompletionPredictor] module from the PowerShell Gallery.
+
```
Comment (StevenBucher98) [inline reference/docs-conceptual/learn/shell/using-predictors.md:144]: We should add a section that links to the creating your own predictor docs.

File: reference/docs-conceptual/learn/shell/using-predictors.md [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+**PSReadLine** 2.2.2 extends the power of Predictive IntelliSense by adding support for plug-in
+modules that use advanced logic to provide suggestions for full commands. The latest version,
+**PSReadLine** 2.2.6, enables predictions by default.
+
+## Using Predictive IntelliSense
+
+When Predictive IntelliSense is enabled, the prediction suggestion appears as colored text following
+the user's cursor. The suggestions from Predictive IntelliSense help new and experienced users of
+PowerShell discover, edit, and execute full commands based on matching predictions. Suggestions can
+come from the user's history and additional domain specific plugins.
```
Comment (StevenBucher98) [inline reference/docs-conceptual/learn/shell/using-predictors.md:20]: We should clarify that users history in this case is different than `Get-History` and is persistent when opening new sessions

File: reference/docs-conceptual/learn/shell/using-predictors.md (hunk 8 above)
Comment (StevenBucher98) [inline reference/docs-conceptual/learn/shell/using-predictors.md:144]: @theJasonHelmick do we want to document that ordering of imported predictors for `ListView`? I would say yes

Comment (sdwheeler) [reply reference/docs-conceptual/learn/shell/using-predictors.md:144]: Oh yeah... I will add a note.

File: reference/docs-conceptual/learn/shell/using-predictors.md (hunk 9 above)
Comment (sdwheeler) [reply reference/docs-conceptual/learn/shell/using-predictors.md:20]: Done

File: reference/docs-conceptual/learn/shell/using-predictors.md (hunk 8 above)
Comment (sdwheeler) [reply reference/docs-conceptual/learn/shell/using-predictors.md:144]: Done

Comment (sdwheeler) [reply reference/docs-conceptual/learn/shell/using-predictors.md:160]: done

**Review by michaeltlombardi — APPROVED** [review]

**Pull request merged**